bindings/go/generator/ocmtypegen/main.go

```diff
@@ -1,114 +1,248 @@
 package main
 
 import (
+	"bufio"
+	"errors"
 	"fmt"
 	"go/ast"
 	"go/parser"
 	"go/token"
-	"log"
+	"log/slog"
 	"os"
 	"path/filepath"
 	"strings"
 )
 
-// GenerateGetTypeMethod generates a GetType method for structs embedding Type
-func GenerateGetTypeMethod(structName string) string {
-	return fmt.Sprintf(`import (
-	"ocm.software/open-component-model/bindings/go/runtime"
+const (
+	typegenMarker = "+ocm:typegen=true"
+	generatedFile = "zz_generated.ocm_type.go"
+	runtimeImport = "ocm.software/open-component-model/bindings/go/runtime"
 )
 
-// GetType returns the type definition of %[1]s as per OCM's Type System.
-// It is the type on which it will be registered with the dynamic type system.
-func (t *%[1]s) GetType() runtime.Type {
-	return t.Type
-}
-`, structName)
+func main() {
+	if len(os.Args) < 2 {
+		slog.Info("Usage: generator <root-folder>")
+		os.Exit(1)
+	}
+	root := os.Args[1]
+
+	packages, err := findGoPackages(root)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error finding Go packages: %v\n", err)
```

Comment:
You're using slog and fmt.Fprintf and slog.Error all over the place. :D Choose one. :D

Reply:
👍 chose slog